src/parseFromAnchor.js

```diff
@@ -12,7 +26,7 @@ function parseFromAnchor(opt_uri) {
 		hostname: link.hostname,
 		password: link.password,
 		pathname: link.pathname[0] === '/' ? link.pathname : '/' + link.pathname,
-		port: link.port,
+		port: extractPortFromURI(opt_uri),
```

Comment:
I guess we need to do differently here. We need to set a flag that tries: 

```js
var link = document.createElement('a');
link.href = 'http://localhost';
const isDefaultPortMisleading = link.port === '0';
if (isDefaultPortMisleading) {...}
```

Then we don't need to extract from string.
wdyt?